tests/ui/feature-gates/feature-gate-pin_ergonomics.rs

```diff
@@ -13,6 +13,10 @@ fn foo(x: Pin<&mut Foo>) {
     let _y: &pin mut Foo = x; //~ ERROR pinned reference syntax is experimental
 }
 
+fn foo_const(x: Pin<&Foo>) {
+    let _y: &pin const Foo = x; //~ ERROR pinned reference syntax is experimental
+}
+
```

Comment:
I'd like to see test coverage that this syntax (`&pin const` / `&pin mut`) isn't accepted in `cfg`ed-out code (e.g. under `#[cfg(any())]`) in either type position or expression position.